Patch-Source: https://github.com/MarcJHuber/event-driven-servers/pull/41
--
From 0730765151429fed3a8966117f04d2a906d00f8e Mon Sep 17 00:00:00 2001
From: Jakub Jirutka <jakub@jirutka.cz>
Date: Mon, 20 Feb 2023 18:32:47 +0100
Subject: [PATCH] mavis: specify SONAME for libmavis.so

This fixes the problem that the tac_plus-ng binary is linked against
libmavis.so (without version), instead of libmavis.so.0.
---
 mavis/Makefile.obj | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mavis/Makefile.obj b/mavis/Makefile.obj
index 766c14c..f8811e1 100644
--- a/mavis/Makefile.obj
+++ b/mavis/Makefile.obj
@@ -102,7 +102,7 @@ $(MAVIS_LIB): $(LIBMAVISOBJS) mavis.h
 
   else
 $(MAVIS_LIB).$(LIBVER_MAJOR).$(LIBVER_MINOR): $(LIBMAVISOBJS) mavis.h
-	$(LD_SHARED) -o $@ $(LIBMAVISOBJS) $(LIB_MAVIS_LIB) $(LIB_DL)
+	$(LD_SHARED) -Wl,-soname,$(MAVIS_LIB).$(LIBVER_MAJOR) -o $@ $(LIBMAVISOBJS) $(LIB_MAVIS_LIB) $(LIB_DL)
 
 $(MAVIS_LIB).$(LIBVER_MAJOR): $(MAVIS_LIB).$(LIBVER_MAJOR).$(LIBVER_MINOR)
 	rm -f $@ ; ln $^ $@
